fusion-engine-driver/utils/conversion_utils.hpp

```diff
@@ -95,5 +103,105 @@ class ConversionUtils {
     pose_stamped.pose.orientation.w = contents.orientation[3];
     return pose_stamped;
   }
-};
 
+  static char calculate_checksum(const std::string& message) {
+    char checksum = 0;
+    for (auto c : message) {
+      checksum ^= c;
+    }
+    return checksum;
+  }
+
+  static std::string ConvertGpsToUtc(double gps_time) {
+    const int kSecondsInDay = 86400;
```

Comment:
This may be simplified if we apply the leap seconds before all the divisions. Also, it is not a good idea to hard code the leap seconds as 18 is only applicable after 2016 and may not be applicable after end of this year. Refer to `UTCTime::UTCTime(const GPSTime& gps_time)` in our code as an example.